### PR TITLE
isaric_washington

### PR DESCRIPTION
**Pull request by ekamau**
initial draft of the CVMIRIJ Washington parser

### CODE REVIEW

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:99]: This needs to be a combined field, the headers are social_history___1, social_history___2 etc. Also in this case we're only looking at smoking - alcohol/substance/unkown should not be filled in. Not sure if we want to include vaping - we haven't up until now.

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:111]: Same here, fields are '*___1, ___2' format

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:169]: This is correct, but because ADMIT DATE and day_to_icu are on different lines in the data file, an error is thrown because one of the fields is always empty when endDate is run. We need to consider how to do this.

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:297]: icu_level_care___7 also relevant here

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:324]: if this date is correct, it should read date = {field = "date_filling_first_form"}. But I think in this case, and a lot of the others, this is the wrong date - I'd suggest 
  date = { field = "ADMIT DATE", apply = { function = "endDate", params = [
    "hospital_day_core_e1",
  ] } }

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:356]: This field is actually just recording the pCO2. replace with 
[[observation]]
  name = "pco2"
  date = { field = "ADMIT DATE", apply = { function = "endDate", params = [
    "hospital_day_core_e1",
  ] } }
  phase = "study"
  value = { field = "pco2_from_same_blood_gas_r" }
  context = ['Same blood gas record']

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:368]: This is recording a value for hco3, which we don't currently have in the schema

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [inline isaric/parsers/isaric_washington.toml:548]: add context 
context = ["Arterial", "Same blood gas record"]

**Review by pipliggins — CHANGES_REQUESTED** [review]
Some things to address - We also need to take another look at how to calculate the dates given that the admission date and durations are usually on different rows in this data.

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [reply isaric/parsers/isaric_washington.toml:137]: I think there's a 28 day mortality field that should be added here

Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:137]: yes, I think there was a 28-day mortality field or entry; wasn't sure how to add.

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:359]: huh?

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:314]: dont think I understand that ..

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:58]: have changed this .. check if now correct.

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:99]: I think I have fixed this ...

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:111]: not sure what you mean .. comorbities___12 is for diabetes. 'comorbidities' is a checkbox ... unless I am wrong ..

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:169]: sure ..

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:324]: ok, nice, will try this

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:368]: Corrected!

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:79]: they are checkbox fields ..

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:99]: not sure if I have corrected this correctly ..

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:284]: yes; that's the spelling in the data

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [reply isaric/parsers/isaric_washington.toml:359]: Have used Pip's suggestions above ..

**Review by pipliggins — APPROVED** [review]

**Comment by abhidg** [comment]
@pipliggins @ekamau let's hold off on merging this parser until we have figured out how to address combining fields from multiple rows in adtl. If this is the only dataset with this feature, can do a one-off transformation script in pandas later.

**Comment by pipliggins** [comment]
I've brought this up to date with all the latest schema changes and edited the data file so it will now run with admission dates on all rows. @sadiekelly I don't think you ever reviewed this parser? It will need looking at before being merged.

File: isaric/parsers/isaric_washington.toml
Comment (sadiekelly) [inline isaric/parsers/isaric_washington.toml:1]: hi @ekamau here's my Washington parser review. The raw data file does not contain data for many of the observation table attributes unfortunately. Let me know if you want to discuss anything further, thanks!
[Washington CVMIRIJ parser review 2023-09-18.xlsx](https://github.com/globaldothealth/isaric/files/12647565/Washington.CVMIRIJ.parser.review.2023-09-18.xlsx)

File: isaric/parsers/isaric_washington.toml
Comment (ekamau) [inline isaric/parsers/isaric_washington.toml:395]: There is another "confusion" above with the same labels and values .. we should remove this.

**Review by ekamau — COMMENTED** [review]
looks good. many thanks @pipliggins .
I will edit to remove the extra "confusion" in the observation table ..

File: isaric/parsers/isaric_washington.toml
Comment (pipliggins) [reply isaric/parsers/isaric_washington.toml]: good catch, thanks @ekamau !

**Pull request merged**
